ppB.py

```diff
@@ -0,0 +1,393 @@
+
+            w.writerow(head)
+            for peak in nin:
+
+                area = peak_sum_area(im, peak)
+                # print('area:', area)
+                peak.set_area(area)
+                ms = peak.get_mass_spectrum()
+                # print("Peaks rt: ", peak.get_rt())
+                # print("Peaks ms_list: ", ms.mass_list)
+                # print("Peaks ms_spec: ", list(ms.mass_spec))
+                p_rt = peak.get_rt()
+                its = []
+                items = list(ms.mass_spec)
+                for i in items:
+                    x = float(i)
+                    its.append(x)
+
+                ms_d = ([area] + [p_rt] + its)
+                # ms_d = its
+                # print('ms_d', ms_d)
+                w.writerow(ms_d)
+
+            f.close()
+
+        peakz.append(nin)
+        # #print("...", counter)
+        # counter += 1
+        ms_data_files.append(completeName)
+    print('ms_data_files:', ms_data_files)
+
+    return [peakz, ms_data_files]
+
+
+def MS_process(file_list):
+    '''
+    @param file_list:
+    @return:
+    '''
+
+    ratio_set = []
+    print(file_list)
+
+    for n in file_list:
+        peaks = []
+        areas = []
+        print('n=', n)
+        print("-------------11------------------------------------")
+        with open(n, 'r') as f:
+            next(f)
+            for line in f:
+                sline = line.split(',')
+                a = sline[0]
+                p = sline[1]
+                # p2 = sline[2]
+                # p3 = sline[3]
+                # print('sline=', sline)
+                # print('are2a=', a)
+                # print('peak rt2=', p)
+
+                sline.pop(0)
+                sline.pop(0)
+                peaks.append(p)
+                areas.append(a)
+
+                ratios = []
+                maxi = max(map(float, sline))
+                # print('maxi=', maxi)
+                # loc = sline.index(str(maxi))
+
+                c = 34.0
+                for i in sline:
+                    r = float(i) / float(maxi)
+                    rx = int(r * 999)
+                    ratios.append([c, rx])
+                    c += 1
+
+                ratio_set.append(ratios)
+            # print('peaks=', peaks)
+            # print('areas=', areas)
+
+            print("----------22--------------------------------------")
+            name1 = n.rsplit('ms_data.csv')
+            name2 = name1[0] + '.txt'
+            print('n2=', name2)
+            # ms_name = os.path.join()
+            # print('ms', name1)
+
+            pp = open(name2, "w+")
+            for ratios, p, a in zip(ratio_set, peaks, areas):
+                # print('ratios=', ratios)
+                rs = sorted(ratios, key=lambda t: t[1], reverse=True)[:10]
+
+                # print('rs=', rs)
+                ll = 'Name:', p, 'Area-', a
+                ll = str(ll).replace("'", "").replace(",", "").replace("(", "").replace(")", "")
```

Comment:
These are repeated and I'd recommend making them into functions.  This code:

```
            pp = open(name2, "w+")
            for ratios, p, a in zip(ratio_set, peaks, areas):
                # print('ratios=', ratios)
                rs = sorted(ratios, key=lambda t: t[1], reverse=True)[:10]

                # print('rs=', rs)
                ll = 'Name:', p, 'Area-', a
                ll = str(ll).replace("'", "").replace(",", "").replace("(", "").replace(")", "")
                # print('ll=', ll)
                pp.write(ll + "\n")

                nn = 'Num Peaks:', 10
                nn = str(nn).replace("'", "").replace(",", "").replace("(", "").replace(")", "")
                # print('nn=', nn)
                pp.write(nn + "\n")
                for i in rs:
                    ss = str(i).replace('[', '').replace(']', '').replace(',', '')
                    # print('ss=', ss)
                    pp.write(ss + "\n")
                # print('\n')
                pp.write("\n")
            pp.close()
```

I would rewrite as something like:

```
            def replace_n_write(ll, pp, chars):
                ll = str(ll)
                for c in chars:
                    ll = ll.replace(c, '')
                pp.write(ll + "\n")

            with open(name2, "w+") as pp:
                for ratios, p, a in zip(ratio_set, peaks, areas):
                    replace_n_write(ll=('Name:', p, 'Area-', a), pp=pp, chars="',()")
                    replace_n_write(ll=('Num Peaks:', 10), pp=pp, chars="',()")

                    for i in sorted(ratios, key=lambda t: t[1], reverse=True)[:10]:
                        replace_n_write(ll=i, pp=pp, chars="[],")
                    pp.write("\n")
```

Also, always try to use the `with open() as f` context manager convention, it's a good habit.

Reply:
`replace_n_write(ll=('Name:', p, 'Area-', a), pp=pp, chars="',()")
`
Is it needed to explicitly state, 'll=...'  or is the position of the variable enough for the script & readers to understand?

If it is for clarity's sake I understand as well. Just curious.